test/utils/pod.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2024 The HAMi Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+)
+
+var Pod = &corev1.Pod{
+	ObjectMeta: metav1.ObjectMeta{
+		Name:      "gpu-pod",
+		Namespace: "default",
+	},
+	Spec: corev1.PodSpec{
+		Containers: []corev1.Container{
+			{
+				Name:    "cuda-container",
+				Image:   "nvcr.io/nvidia/k8s/cuda-sample:vectoradd-cuda12.5.0",
+				Command: []string{"/bin/sh"},
+				Args:    []string{"-c", "sleep 86400"},
```

Comment:
I suggest writing it like this here.

```
nvidia-smi -L; trap 'exit' TERM; sleep infinity & wait
```

Reply:
@chaunceyjiang this section is designed to align closely with the example https://github.com/Project-HAMi/HAMi/blob/d04fc8baaad6b9688511ec5158cfdc6fd6832cee/examples/nvidia/example.yaml#L9